lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -198,7 +195,9 @@ impl<TInvertedIndex: InvertedIndex> VectorIndex for SparseVectorIndex<TInvertedI
             ram_index.upsert(id, vector.to_owned());
             index_point_count += 1;
         }
-        self.max_point_id = index_point_count.saturating_sub(1) as PointOffsetType;
+        // the underlying upsert operation does not guarantee that the indexed vector count is correct
+        // so we set the indexed vector count to the number of points we have seen
+        ram_index.indexed_vector_count = index_point_count;
```

Comment:
TBH, I don't like this approach. `ram_index.upsert` updates already this count, why do we need set it again?